orchagent/portsorch.cpp

```diff
@@ -448,191 +466,460 @@ PortsOrch::PortsOrch(DBConnector *db, DBConnector *stateDb, vector<table_name_wi
+            attr.value.booldata = cit.autoneg.value;
+            attrList.push_back(attr);
+        }
 
-    if (sai_query_attribute_enum_values_capability(gSwitchId, SAI_OBJECT_TYPE_VLAN,
-                                                   SAI_VLAN_ATTR_UNKNOWN_UNICAST_FLOOD_CONTROL_TYPE,
-                                                   &values) != SAI_STATUS_SUCCESS)
+        if (cit.adv_speeds.is_set)
+        {
+            PortAttrValue_t attrValue;
+            auto &outList = attrValue.adv_speeds;
+            auto &inList = cit.adv_speeds.value;
+            outList.insert(outList.begin(), inList.begin(), inList.end());
+            attrValueList.push_back(attrValue);
+
+            attr.id = SAI_PORT_ATTR_ADVERTISED_SPEED;
+            attr.value.u32list.count = static_cast<std::uint32_t>(attrValueList.back().adv_speeds.size());
+            attr.value.u32list.list = attrValueList.back().adv_speeds.data();
+            attrList.push_back(attr);
+        }
+
+        if (cit.interface_type.is_set)
+        {
+            attr.id = SAI_PORT_ATTR_INTERFACE_TYPE;
+            attr.value.s32 = cit.interface_type.value;
+            attrList.push_back(attr);
+        }
+
+        if (cit.adv_interface_types.is_set)
+        {
+            PortAttrValue_t attrValue;
+            auto &outList = attrValue.adv_interface_types;
+            auto &inList = cit.adv_interface_types.value;
+            outList.insert(outList.begin(), inList.begin(), inList.end());
+            attrValueList.push_back(attrValue);
+
+            attr.id = SAI_PORT_ATTR_ADVERTISED_INTERFACE_TYPE;
+            attr.value.s32list.count = static_cast<std::uint32_t>(attrValueList.back().adv_interface_types.size());
+            attr.value.s32list.list = attrValueList.back().adv_interface_types.data();
+            attrList.push_back(attr);
+        }
+
+        if (cit.fec.is_set)
+        {
+            attr.id = SAI_PORT_ATTR_FEC_MODE;
+            attr.value.s32 = cit.fec.value;
+            attrList.push_back(attr);
+        }
+
+        if (cit.mtu.is_set)
+        {
+            attr.id = SAI_PORT_ATTR_MTU;
+            /* mtu + 14 + 4 + 4 = 22 bytes */
+            attr.value.u32 = cit.mtu.value + (uint32_t)(sizeof(struct ether_header) + FCS_LEN + VLAN_TAG_LEN);
+            attrList.push_back(attr);
+        }
+
+        if (cit.tpid.is_set)
+        {
+            attr.id = SAI_PORT_ATTR_TPID;
+            attr.value.u16 = cit.tpid.value;
+            attrList.push_back(attr);
+        }
+
+        if (cit.pfc_asym.is_set)
+        {
+            attr.id = SAI_PORT_ATTR_PRIORITY_FLOW_CONTROL_MODE;
+            attr.value.s32 = cit.pfc_asym.value;
+            attrList.push_back(attr);
+        }
+
+        attrDataList.push_back(attrList);
+        attrCountList.push_back(static_cast<std::uint32_t>(attrDataList.back().size()));
+        attrPtrList.push_back(attrDataList.back().data());
+    }
+
+    auto status = sai_port_api->create_ports(
```

Comment:
From HLD, what if create_ports is NULL or not supported?

<img width="635" alt="image" src="https://user-images.githubusercontent.com/45705344/213960643-ce58ceb6-f7cc-455c-bd34-2b389d419d7f.png">

Reply:
@prgeor these statements are not relevant:
1. Checking for NULL doesn't make any sense - the bulk API (sairedis) is implemented and currently available for all the vendors
2. The current syncd implementation doesn't propagate `SAI_STATUS_NOT_SUPPORTED` and `SAI_STATUS_NOT_IMPLEMENTED` codes to upper layer. Instead, it automatically falls back to the legacy one-by-one processing of the requested bulk operation objects. Taking this into consideration, there is no point to have a capabilities check in swss